redwood-widget/src/androidMain/kotlin/app/cash/redwood/widget/SavedStateRegistry.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood.widget
+
+import android.os.Binder
+import android.os.Bundle
+import android.os.Parcelable
+import android.util.Size
+import android.util.SizeF
+import android.util.SparseArray
+import androidx.savedstate.SavedStateRegistry as AndroidXSavedStateRegistry
+import java.io.Serializable
+
+internal class AndroidSavedStateRegistry(
+  private val key: String,
+  private val delegate: AndroidXSavedStateRegistry,
+) : SavedStateRegistry {
+  override fun canBeSaved(value: Any): Boolean {
+    // lambdas in Kotlin implement Serializable, but will crash if you really try to save them.
+    // we check for both Function and Serializable (see kotlin.jvm.internal.Lambda) to support
+    // custom user defined classes implementing Function interface.
```

Comment:
Nice comment

Reply:
This file needs an AOSP header! Can't take credit